### PR TITLE
Added protocol handling for ServicePort and fixed type/value naming inside generic receiver

### PR DESCRIPTION
**Pull request by oldium**
**Description:**

Fixes three bugs:

1. The "type/value" naming format was wrongly handled inside common receiver code (used by generic receiver)
2. The Protocol (TCP/UDP) was not detected when creating K8s ServicePort
3. The same port number could not be exposed with different protocol

**Link to tracking Issue:**

resolves: #2597
resolves:  #767

**Testing:**

Added tests and also tested `udplog` and `tcplog` in our deployment.

**Documentation:**

This is purely a bugfix, so no documentation.

### CODE REVIEW

**Comment by oldium** [comment]
Changelog pipeline failed, I will add an entry mentioning fix of the #2597

**Comment by oldium** [comment]
Two changelog entries added. Ready.

**Comment by oldium** [comment]
Working on lint...

**Comment by oldium** [comment]
Changelog done, Lint done

File: internal/manifests/collector/parser/receiver/receiver.go [hunk 1]
```diff
@@ -104,30 +104,12 @@ func isScraperReceiver(name string) bool {
 
 func singlePortFromConfigEndpoint(logger logr.Logger, name string, config map[interface{}]interface{}) *v1.ServicePort {
 	var endpoint interface{}
+	var _type = receiverType(name)
```
[the remark this answers is not in the document]
Comment (oldium) [reply internal/manifests/collector/parser/receiver/receiver.go:107]: Yes, easy :-)

File: internal/manifests/collector/parser/receiver/receiver_syslog_test.go [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+package receiver
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+)
+
+func TestSyslogSelfRegisters(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (oldium) [reply internal/manifests/collector/parser/receiver/receiver_syslog_test.go:24]: Ok, I will merge TestSyslogNoConfiguration/Udp*/Tcp* into a test table.

**Comment by oldium** [comment]
Done. Anything more?

**Comment by oldium** [comment]
I see that [End-to-end tests e2e on K8s 1.29](https://github.com/open-telemetry/opentelemetry-operator/actions/runs/7886194453/job/21561034422#logs) test failed, but I cannot find anything specific to this Pull Request.

File: internal/manifests/collector/parser/receiver/receiver.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (oldium) [reply internal/manifests/collector/parser/receiver/receiver.go]: Done, force-pushed the change.

File: internal/manifests/collector/parser/receiver/receiver_syslog_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (oldium) [reply internal/manifests/collector/parser/receiver/receiver_syslog_test.go:24]: Done, force-pushed the change.

**Review by pavolloffay — APPROVED** [review]

**Review by jaronoff97 — APPROVED** [review]

**Comment by jaronoff97** [comment]
thank you for your contribution!

**Pull request merged**